security/pkg/nodeagent/sds/sdsservice_test.go

```diff
@@ -469,8 +470,8 @@ func TestStreamSecretsMultiplePush(t *testing.T) {
 	defer server.Stop()
 
 	conn, stream := createSDSStream(t, socket)
-	proxyID := "sidecar~127.0.0.1~id2~local"
 	defer conn.Close()
+	proxyID := "sidecar~127.0.0.1~id5~local"
 	notifyChan := make(chan string)
 	go testSDSStreamMultiplePush(t, stream, proxyID, notifyChan)
```

Comment:
You cannot fail a test from a goroutine, See https://golang.org/pkg/testing/#B.FailNow

Reply:
Thanks. Replaced testing.T with a channel to push errors into main goroutine.